legacy_publisher/dita_publisher.py

```diff
@@ -258,3 +284,115 @@ def create_collection_page(classlist=None,export_dita=None):
     return root
 
 
+
+def create_class_page(class_data=None,export_dita=None):
+    root = create_dita_root(doctype_str=None)
+    title = os.path.basename(str(export_dita)).replace(".dita","")
+    topic = create_class(root=root,id=title)
+    body = create_body(root=root,topic=topic)
+    images = create_images(root=root,body=body)
+
+    if len(class_data.summary.table) != 0 : 
+        summary = create_section(root=root,body=body, section=str(class_data.summary.title), title_str=None)
+        tbody = create_classtable(root=root,section=summary, cols=str(class_data.colspan))
+        process_section(class_data.summary.table,root,tbody)
+
+    if len(class_data.signatures.table) != 0 : 
+        signatures = create_section(root=root,body=body, section=str(class_data.signatures.title), title_str=str(class_data.signatures.title))
+        tbodysignatures = create_classtable(root=root,section=signatures, cols=str(class_data.colspan))
+        process_complex_section(class_data.signatures.table,root,tbodysignatures, class_data.colspan, 4)
+
```

Comment:
it _looks-like_ the next two blocks  (propulsion and remarks) both rely on there being a table in the content.  I _think_ the actual logic is to see if the elements are non-null.